lineage/states/stateCommon.py

```diff
@@ -130,21 +134,27 @@ def gamma_estimator_atonce(gamma_obs, time_cen, gamas, x0=None):
     arrgs = (arg1, arg2, arg3, arg4)
 
     if x0 is None:
-        x0 = np.array([20.0, 2.0, 3.0, 4.0, 5.0])
-
-    A = np.zeros((3, 5))
-    np.fill_diagonal(A[:, 1:], -1.0)
-    np.fill_diagonal(A[:, 2:], 1.0)
+        if phase:
+            x0 = np.array([20.0, 2.0, 3.0, 4.0, 5.0])
+        else:
+            x0 = np.array([20.0, 2.0, 3.0])
 
-    # Override x0 if we were given a bad starting point
-    if np.allclose(np.dot(A, x0), 0.0):
-        x0 = np.array([20.0, 1.0, 2.0, 3.0, 4.0])
+    if phase: # phase-specific observations
+        A = np.zeros((3, 5)) # is a matrix that contains the constraints. the number of rows shows the number of linear constraints.
+        np.fill_diagonal(A[:, 1:], -1.0)
+        np.fill_diagonal(A[:, 2:], 1.0)
+        linc = LinearConstraint(A, lb=np.zeros(3), ub=np.ones(3) * 100.0)
+        if np.allclose(np.dot(A, x0), 0.0):
+            x0 = np.array([20.0, 1.0, 2.0, 3.0, 4.0])
 
-    linc = LinearConstraint(A, lb=np.zeros(3), ub=np.ones(3) * 100.0)
-    bnds = Bounds(lb=np.ones_like(x0) * 0.001, ub=np.ones_like(x0) * 100.0, keep_feasible=True)
+    bnds = Bounds(lb=np.ones_like(x0) * 0.01, ub=np.ones_like(x0) * 100.0, keep_feasible=True)
     HH = BFGS()
 
-    res = minimize(nLL_atonceJ, x0=x0, jac=True, hess=HH, args=arrgs, method="trust-constr", bounds=bnds, constraints=[linc])
-    assert (res.success is True) or ("maximum number of function evaluations is exceeded" in res.message)
+    if phase:
+        res = minimize(nLL_atonceJ, x0=x0, jac=True, hess=HH, args=arrgs, method="trust-constr", bounds=bnds, constraints=[linc])
```

Comment:
It'd be much better to make an if statement that sets a variable containing the `constraints` argument, and then move these two more complex lines out of the if statement so they're not generally duplicated.